### PR TITLE
#181414705 (chore) CI testing with Github actions 

### PR DESCRIPTION
**Pull request by niyojuste**
### What does this PR do?
 Sets up continuous integration testing
### Description of Task to be completed?
Configure a Github workflow that runs test for every pull request into develop or main branches
### How should this be manually tested?
Get started By Running commands bellow:
  - clone repo
  - checkout to your new branch
  - Add your changes
  - Push to your new remote branch
  - Create a new pull request to develop
  - At the bottom of your pull request it should be running the tests
  - Await to see if the checks pass
 
### Any background context you want to provide?
Configured with coveralls. A badge on the README shall read the coverage of all the tests
### What are the relevant pivotal tracker stories?
[##181414705](https://www.pivotaltracker.com/story/show/181414705)
### Questions:

### CODE REVIEW

**Review by ericus123 — COMMENTED** [review]
Add a step for running tests

**Review by Procech20 — APPROVED** [review]
LGTM

**Review by Robertishimwe — APPROVED** [review]

**Review by itisWasp — CHANGES_REQUESTED** [review]
Please check your codes to many files changed because of formatting please check if you have prettier configured and installed as this will affect us when we are rebasing since all our code base are formatted by prettier.

File: test/usertest.spec.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+                .request(app)
+                .get('/api/v2/users')
+                .end((err, res) => {
+                    res.should.have.status(200);
+                    done();
+                })
+                .timeout(100000);
+        });
+    });
+});
```
Comment (didasmbalanya) [inline test/usertest.spec.js]: This should fail unless for the right users. Look for a dummy endpoint that should not be protected and can test a call to the db. Or do a direct test with the db. Create something and read it

**Review by didasmbalanya — CHANGES_REQUESTED** [review]

**Comment by ericus123** [comment]
Add test script in pre-commit hook

File: test/usertest.spec.js [hunk 2]
```diff
@@ -1,29 +1,36 @@
...
+//         .request(app)
+//         .get('/api/v2/users')
+//         .end((err, res) => {
+//           res.should.have.status(200);
+//           done();
+//         })
+//         .timeout(100000);
+//     });
+//   });
+// });
```
Comment (ericus123) [inline test/usertest.spec.js]: This should be removed

Comment (niyojuste) [reply test/usertest.spec.js]: Now removed

Comment (ericus123) [reply test/usertest.spec.js]: Also you can add a CI(Github actions) status badge

**Review by ericus123 — APPROVED** [review]

**Review by itisWasp — APPROVED** [review]

**Pull request merged**